src/FsCheck/Arbitrary.fs

```diff
@@ -793,6 +799,11 @@ module Arb =
                             yield new KeyValuePair<_,_>(key, value) }
             fromGenShrink(genKeyValuePair,shrinkKeyValuePair)
 
+        static member NegativeInt() =
+            Default.Int32()
+            |> mapFilter ((*) -1) (fun x -> x < 0)
```

Comment:
This should be `|> mapFilter (fun x -> -abs x) (fun x -> x < 0)` ideally. 

The map (first function) is really an optimisation - the filter (second function) is applied regardless so even if the map doesn't work you won't see a problem in the generated values. But a good map decreases the number of values that are being generated then thrown away by the filter.

Your map function just flips the sign, so it isn't very effective in making a positive or negative int into a negative int. My suggestion above makes them all negative except 0s (which will still be filtered out by the filter)